### PR TITLE
remove runtime crate

### PR DESCRIPTION
**Pull request by leonardoalt**
For the tests that use crates it's possible to use the same powdr-riscv-rt crate from the checkout. For file tests we need to use git. I kept them separate for now but I can make everything git if that's preferred.

### CODE REVIEW

File: riscv/src/lib.rs [hunk 1]
```diff
@@ -152,7 +152,7 @@ version = "0.1.0"
 edition = "2021"
 
 [dependencies]
-runtime = {{ path = "./runtime" }}
+powdr_riscv_rt = {{ git = "https://github.com/powdr-labs/powdr", branch = "main" }}
```
Comment (chriseth) [inline riscv/src/lib.rs:155]: Doesn't this also need the `runtime` subdirectory?

File: riscv/tests/riscv_data/vec_median/Cargo.toml [hunk 2]
```diff
@@ -4,6 +4,6 @@ version = "0.1.0"
 edition = "2021"
 
 [dependencies]
-runtime = { path = "../../../runtime" }
+powdr-riscv-rt = { path = "../../../../powdr_riscv_rt" }
```
Comment (chriseth) [inline riscv/tests/riscv_data/vec_median/Cargo.toml:7]: I would opt for actually spelling `runtime` out. `rt` can mean so many things.

File: riscv/src/lib.rs (hunk 1 above)
Comment (leonardoalt) [reply riscv/src/lib.rs:155]: What for?

File: riscv/tests/riscv_data/vec_median/Cargo.toml (hunk 2 above)
Comment (leonardoalt) [reply riscv/tests/riscv_data/vec_median/Cargo.toml:7]: I think this is rather common. We use another crate called riscv_rt for qemu, so i thought powdr-riscv-rt fits well

Comment (leonardoalt) [reply riscv/tests/riscv_data/vec_median/Cargo.toml:7]: I mean, common also if you include the context you get from the "riscv" prefix to "rt"

Comment (chriseth) [reply riscv/tests/riscv_data/vec_median/Cargo.toml:7]: Even if it's common, I really don't think it's a good idea. We also don't abbreviate executor or other stuff.

Comment (chriseth) [reply riscv/tests/riscv_data/vec_median/Cargo.toml:7]: I mean isn't even runtime already an abbreviation for runtime environment?

Comment (leonardoalt) [reply riscv/tests/riscv_data/vec_median/Cargo.toml:7]: Ok. I still have a somewhat strong opinion about this. `powdr-riscv-runtime` is just really long and kinda clumsy. If someone else says they prefer it spelled out I'll give in.

Comment (chriseth) [reply riscv/tests/riscv_data/vec_median/Cargo.toml:7]: What does it matter if it's long and clumsy? Do you need to type it that often?

Comment (leonardoalt) [reply riscv/tests/riscv_data/vec_median/Cargo.toml:7]: just aesthetics

**Review by chriseth — APPROVED** [review]

**Pull request merged**